src/pages/iou/request/step/IOURequestStepRate.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
+import SelectionList from '@components/SelectionList';
+import RadioListItem from '@components/SelectionList/RadioListItem';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as IOU from '@libs/actions/IOU';
+import compose from '@libs/compose';
+import DistanceRequestUtils from '@libs/DistanceRequestUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Route} from '@src/ROUTES';
+import type {Policy} from '@src/types/onyx';
+import StepScreenWrapper from './StepScreenWrapper';
+import withWritableReportOrNotFound from './withWritableReportOrNotFound';
+
+type Props = {
+    /** Object of last selected rates for the policies */
+    lastSelectedDistanceRates: Record<string, string>;
+
+    /** Policy details */
+    policy: OnyxEntry<Policy>;
+
+    /** The route object passed to this screen */
+    route: {
+        /** The params passed to this screen */
+        params: {
+            /** The route to go back to */
+            backTo: Route;
+        };
+    };
+};
+
+function IOURequestStepRate({
+    policy,
+    route: {
+        params: {backTo},
+    },
+    lastSelectedDistanceRates = {},
+}: Props) {
+    const styles = useThemeStyles();
+    const {translate, toLocaleDigit} = useLocalize();
+    const rates = DistanceRequestUtils.getMileageRates(policy?.id);
```

Comment:
Let's use withOnyx to pass a rates prop, and pass the policy object to the selector.

Reply:
why we would need to do it this way?

Reply:
It's more of a general principle vs an actual problem right now. I do think it's bad to pass the policyID to that function and access the policy within it, when we could just pass the policy directly.

We want to have data come from props and state directly. Accessing data like this makes it harder to debug, and it also presents the opportunity for bugs. Often people use a util like this to get a value, and that value might not be updated when the underlying object is, because it's possible that the component doesn't update when the value being accessed changes. That's not happening here, but it's part of why we have this recommendation. Here is the issue where we refactored a bunch of stuff due to this problem https://github.com/Expensify/App/issues/27262

Also, it's more performant to access only the data needed from withOnyx and using a selector to get the exact value, vs subscribing to every change of the key.

Reply:
It's more of a general principle vs an actual problem right now. I do think it's bad to pass the policyID to that function and access the policy within it, when we could just pass the policy directly.

We want to have data come from props and state directly. Accessing data like this makes it harder to debug, and it also presents the opportunity for bugs. Often people use a util like this to get a value, and that value might not be updated when the underlying object is, because it's possible that the component doesn't update when the value being accessed changes. That's not happening here, but it's part of why we have this recommendation. Here is the issue where we refactored a bunch of stuff due to this problem https://github.com/Expensify/App/issues/27262

Also, it's more performant to access only the data needed from withOnyx and using a selector to get the exact value, vs subscribing to every change of the key.